cmd/veneur-emit/main.go

```diff
@@ -170,30 +230,131 @@ func sendSpan(conn MinimalConn, span *ssf.SSFSpan) error {
 	return nil
 }
 
-func sendMetrics(client MinimalClient, passedFlags map[string]bool, name string, tags []string) error {
+// TODO: validity checking? or just fire and forget
+func buildEventPacket(passedFlags map[string]flag.Value) (bytes.Buffer, error) {
+	var buffer bytes.Buffer
+	buffer.WriteString("_e")
+
+	if passedFlags["e_title"] == nil {
+		return bytes.Buffer{}, errors.New("missing event title")
+	}
+
+	if passedFlags["e_text"] == nil {
+		return bytes.Buffer{}, errors.New("missing event text")
+	}
+
+	buffer.WriteString(fmt.Sprintf("{%d,%d}", len(passedFlags["e_title"].String()), len(passedFlags["e_text"].String())))
+	buffer.WriteString(":")
+
+	buffer.WriteString(passedFlags["e_title"].String())
+	buffer.WriteString("|")
+
+	buffer.WriteString(passedFlags["e_text"].String())
+
+	if passedFlags["e_time"] != nil {
+		buffer.WriteString(fmt.Sprintf("|d:%s", passedFlags["e_time"].String()))
```

Comment:
Ignorable suggestion: Make a slice of `thing:value` strings and use `strings.Join('|', …)`?